consensus/state.go

```diff
@@ -388,6 +392,7 @@ func (cs *ConsensusState) SetProposalAndBlock(proposal *types.Proposal, block *t
 
 func (cs *ConsensusState) updateHeight(height int64) {
 	cs.Height = height
+	cs.metrics.Height.Add(float64(height - cs.Height))
```

Comment:
Same as with logging, we keep interleaving non-business concerns with our business logic, which makes for harder to maintain code. I know this would require a bigger refactor but this might be a good time to think about better composability. Which could be done with the more rigirous use of interfaces and then layered decoratored implementations for the different concerns (logic, logging, metrics). This [video](https://www.youtube.com/watch?v=xyDkyFjzFVc) gives an intro how that looks and I can give some examples how it is done in production code.

Reply:
Video is great. I love it.